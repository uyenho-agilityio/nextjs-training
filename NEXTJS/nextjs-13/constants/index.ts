export const LOGO = {
  alt: 'Logo',
  src: '/images/logo.svg',
  width: 32,
  height: 32,
};

export const PAGE_404 = {
  src: '/images/404.jpg',
  alt: '404 Page',
  width: 400,
  height: 400,
};

export const PAGE_ROUTES = {
  HOME: '/',
  ABOUT: '/about',
  EVENTS: '/events',
};

export const ROUTE_LIST = [
  {
    label: 'Home',
    path: '/',
  },
  {
    label: 'Events',
    path: '/events',
  },
  {
    label: 'About',
    path: '/about',
  },
];

export const MESSAGES = {
  NETWORK_ERROR: 'Network error. Please try again later.',
  ERROR: 'Something went wrong!',
};

export const TEXT = {
  CONTINUE: 'Continue',
  BACK_HOME: 'BACK TO HOME',
  HEAD_BACK: "Let's Head Back!",
  SIGN_IN: 'Sign in',
  ACADEMY: 'Academy',
};

export const SEO_DATA = {
  title: 'NextJS Training with app/ directory',
  description: 'Do examples to understand NextJS 13',
  icon: '/favicon.ico',
  url: 'https://nextjs-examples-training.vercel.app/',
  thumbnailUrl:
    'https://upload.wikimedia.org/wikipedia/commons/thumb/8/8e/Nextjs-logo.svg/1200px-Nextjs-logo.svg.png',
};

export const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
export const LOCAL_BASE_URL = process.env.NEXT_PUBLIC_LOCAL_BASE_URL;
