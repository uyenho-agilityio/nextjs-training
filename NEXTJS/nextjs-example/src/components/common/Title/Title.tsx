// Libs
import { Heading, HeadingProps } from '@chakra-ui/react';
import React, { ReactNode } from 'react';

export interface TitleProps extends HeadingProps {
  size?: string;
  variant?: string;
  children: ReactNode;
}

export const Title: React.FC<TitleProps> = ({ size, variant, children, ...props }): JSX.Element => {
  return (
    <Heading size={size} variant={variant} {...props}>
      {children}
    </Heading>
  );
};
