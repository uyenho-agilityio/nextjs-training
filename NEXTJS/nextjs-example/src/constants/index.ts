export const VERCEL_LOGO = {
  alt: 'Vercel Logo',
  src: '/images/vercel.svg',
  width: 72,
  height: 16,
};

export const COURSE_LOGO = {
  alt: 'Logo',
  src: '/images/logo.svg',
  width: 32,
  height: 32,
};

export const AVATAR = {
  alt: 'Avatar',
  src: '/images/avatar.jpg',
  width: 40,
  height: 40,
};

export const PAGE_404 = {
  alt: '404 Page',
  src: '/images/404.jpg',
  width: 400,
  height: 400,
};

export const PAGE_ROUTES = {
  HOME: '/',
  ABOUT: '/about',
  EVENTS: '/events',
};

export const MESSAGES = {
  NETWORK_ERROR: 'Network error. Please try again later.',
  ERROR: 'Something went wrong!',
};

export const TEXT = {
  CONTINUE: 'Continue',
  BACK_TO_COURSE: 'Back to Course',
  BACK_HOME: 'BACK TO HOME',
  HEAD_BACK: "Let's Head Back!",
  SIGN_IN: 'Sign in',
  ACADEMY: 'Academy',
};
