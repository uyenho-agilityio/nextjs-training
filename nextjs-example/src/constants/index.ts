export const VERCEL_LOGO = {
  alt: 'Vercel Logo',
  src: '/images/vercel.svg',
  width: 72,
  height: 16,
};

export const COURSE_LOGO = {
  alt: 'Logo',
  src: '/images/logo.svg',
  width: 32,
  height: 32,
};

export const AVATAR = {
  alt: 'Avatar',
  src: '/images/avatar.jpg',
  width: 40,
  height: 40,
};

export const PAGE_404 = {
  alt: '404 Page',
  src: '/images/404.jpg',
  width: 400,
  height: 400,
};

export const PAGE_ROUTES = {
  HOME: '/',
  ABOUT: '/about',
  EVENTS: '/events',
};

export const ROUTE_LIST = [
  {
    label: 'Home',
    path: '/',
  },
  {
    label: 'Events',
    path: '/events',
  },
  {
    label: 'About',
    path: '/about',
  },
  {
    label: 'NextJS',
    path: '/nextjs',
  },
];

export const MESSAGES = {
  NETWORK_ERROR: 'Network error. Please try again later.',
  ERROR: 'Something went wrong!',
};

export const TEXT = {
  CONTINUE: 'Continue',
  BACK_TO_COURSE: 'Back to Course',
  BACK_HOME: 'BACK TO HOME',
  HEAD_BACK: "Let's Head Back!",
  SIGN_IN: 'Sign in',
  ACADEMY: 'Academy',
};

export const SEO_DATA = {
  title: 'NextJS Training',
  description: 'Do examples to understand NextJS',
  icon: '/images/favicon.ico',
  url: 'https://nextjs-examples-training.vercel.app/',
  thumbnailUrl:
    'https://upload.wikimedia.org/wikipedia/commons/thumb/8/8e/Nextjs-logo.svg/1200px-Nextjs-logo.svg.png',
};

export const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
