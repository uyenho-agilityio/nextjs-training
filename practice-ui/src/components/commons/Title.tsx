import React, { ReactNode } from 'react';
import { TitleVariants } from '~enums/variant';

export interface TitleProps {
  className?: string;
  fs?: string;
  lh?: string;
  children: ReactNode;
  variant?: TitleVariants;
}

export const Title: React.FC<TitleProps> = React.memo(
  ({ className, fs, lh, children, variant = TitleVariants.Title }): JSX.Element => (
    <div
      className={`${variant === TitleVariants.Card ? `card-title` : ''} ${className}`}
      style={{
        fontSize: fs,
        lineHeight: lh,
      }}
    >
      {children}
    </div>
  ),
);
