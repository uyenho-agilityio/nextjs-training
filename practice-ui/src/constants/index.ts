export * from './footer';
export * from './header';
export * from './management';
export * from './pricing';
export * from './product';
