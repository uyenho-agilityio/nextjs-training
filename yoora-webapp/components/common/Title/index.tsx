'use client';

// Lib
import { Heading, HeadingProps } from '@chakra-ui/react';

type TitleProps = {
  size?: string;
  variant?: string;
} & HeadingProps;

export const Title: React.FC<TitleProps> = ({ size, variant, ...props }): JSX.Element => (
  <Heading size={size} variant={variant} {...props} />
);
