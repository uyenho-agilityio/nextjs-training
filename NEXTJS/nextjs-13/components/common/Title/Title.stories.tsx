// Lib
import Storybook from '@storybook/react';

// Components
import { Title } from './Title';

// Constants
import { TEXT } from '@webapp/constants';

export default {
  title: 'Components/Title',
  component: Title,
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as Storybook.ComponentMeta<typeof Title>;

const Template: Storybook.ComponentStory<typeof Title> = args => <Title {...args} />;

export const Academy = Template.bind({});
Academy.args = {
  children: TEXT.ACADEMY,
};
