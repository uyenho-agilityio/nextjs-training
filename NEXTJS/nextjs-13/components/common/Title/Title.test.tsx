// Lib
import { render, screen } from '@testing-library/react';

// Component
import { Title } from './Title';

// Constant
import { TEXT } from 'constants/index';

describe('Title render', () => {
  test('should render title component with default props', () => {
    const { container } = render(<Title>{TEXT.ACADEMY}</Title>);
    expect(container).toMatchSnapshot();
  });

  test('should render title component with valid props', () => {
    render(<Title>{TEXT.ACADEMY}</Title>);
    expect(screen.getByText(/academy/i)).toBeTruthy();
  });
});
