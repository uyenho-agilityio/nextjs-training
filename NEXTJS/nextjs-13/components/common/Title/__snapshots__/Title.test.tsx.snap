// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title render should render title component with default props 1`] = `
<div>
  <h2
    class="chakra-heading css-0"
  >
    Academy
  </h2>
</div>
`;
